packages/core/src/authoring/queries.ts

```diff
@@ -0,0 +1,159 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ProviderToken} from '../di';
+import {Signal} from '../render3/reactivity/api';
+
+function viewChildFn<V>(locator: ProviderToken<V>|string): Signal<V|undefined>;
+function viewChildFn<V, T>(
+    locator: ProviderToken<T>|string, opts: {read: ProviderToken<V>}): Signal<V|undefined>;
+function viewChildFn<V, T>(
+    locator: ProviderToken<T>|string, opts?: {read?: ProviderToken<V>}): Signal<V|undefined> {
+  return null! as Signal<V|undefined>;
+}
+
+function viewChildRequiredFn<V>(locator: ProviderToken<V>|string): Signal<V>;
+function viewChildRequiredFn<V, T>(
+    locator: ProviderToken<T>|string, opts: {read: ProviderToken<V>}): Signal<V>;
+function viewChildRequiredFn<V, T>(
+    locator: ProviderToken<T>|string, opts?: {read?: ProviderToken<V>}): Signal<V> {
+  return null! as Signal<V>;
+}
+
+viewChildFn.required = viewChildRequiredFn;
+
+/**
+ * Type of the `viewChild` function.
+ *
+ * The viewChild function creates a singular view query. It is a special function that also provides
+ * access to required query results via the `.required` property.
+ */
+export type ViewChildFunction = typeof viewChildFn&{required: typeof viewChildRequiredFn};
+
+/**
+ * Initializes a view child query. Consider using `viewChild.required` for queries that should
+ * always match.
+ *
+ * @usageNotes
+ * Create a child query in your component by declaring a
+ * class field and initializing it with the `viewChild()` function.
+ *
+ * ```ts
+ * @Component({template: '<div #el></div><my-component #cmp />'})
+ * export class TestComponent {
+ *   divEl = viewChild<ElementRef>('el');                   // Signal<ElementRef|undefined>
+ *   divElRequired = viewChild<ElementRef>.required('el');  // Signal<ElementRef>
```

Comment:
```suggestion
 *   divElRequired = viewChild.required<ElementRef>('el');  // Signal<ElementRef>
```

In order to stay consistent with how we set the type on required inputs, should we set the type in the required fn instead of viewChild?

This way it also matches the test cases added below.